File: imstargg-frontend/src/model/PlayerBattleEvent.ts
import {BattleEventMode} from "@/model/enums/BattleEventMode";
import {BattleEventMap} from "@/model/BattleEventMap";

export interface PlayerBattleEvent {
  id: number | null;
  mode: BattleEventMode | null;
  map: BattleEventMap;
}

File: core/core-domain/src/main/java/com/imstargg/core/domain/PlayerBattleEvent.java
package com.imstargg.core.domain;

import com.imstargg.core.domain.brawlstars.BattleEventMap;
import com.imstargg.core.enums.BattleEventMode;
import jakarta.annotation.Nullable;

public record PlayerBattleEvent(
        @Nullable BrawlStarsId id,
        @Nullable BattleEventMode mode,
        BattleEventMap map
) {
}
